packages/server/src/analytics/storage.ts

```diff
@@ -0,0 +1,69 @@
+import { FirestoreService } from '../service/firestore'
+import { analyticsLogic } from './logic'
```

Comment:
The problem with this is that it requires us to call `initializeOrSync` multiple times.  We should only make the minimum necessary DB call -- they are expensive.  So let's create a private bool `synced = false` and set it to true once you call `initializeOrSync`.

Calling `isFirstQuery` or `data` should throw if `synced` is false.

The user is then required to call `initializeOrSync` manually before using it.  That way, it only needs to be called once.